hack/cluster.sh

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env bash
+#
+# This file is part of the Kepler project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Copyright 2022 The Kepler Contributors
+#
+set -eu -o pipefail
+
+# NOTE: assumes that the project root is one level up
+PROJECT_ROOT=$(cd "$(dirname "${BASH_SOURCE[0]}")/.." >/dev/null 2>&1 && pwd)
+declare -r PROJECT_ROOT
+
+# NOTE: this allows common settings to be stored as `.env` file
+# shellcheck disable=SC1091
+source "$PROJECT_ROOT/.env" || true
+
+declare -r TMP_DIR="$PROJECT_ROOT/tmp"
+declare -r LOCAL_DEV_CLUSTER_DIR="$TMP_DIR/local-dev-cluster"
+declare -r LOCAL_DEV_CLUSTER_VERSION="${LOCAL_DEV_CLUSTER_VERSION:-v0.0.3}"
+
+# Supported CLUSTER_PROVIDER are kind,microshift
+export CLUSTER_PROVIDER=${CLUSTER_PROVIDER:-kind}
+
+clone_local_dev_cluster() {
+	if [ -d "$LOCAL_DEV_CLUSTER_DIR" ]; then
+		echo "using local local-dev-cluster"
+		return 0
+	fi
+
+	echo "downloading local-dev-cluster"
+	git clone -b "$LOCAL_DEV_CLUSTER_VERSION" \
+		https://github.com/sustainable-computing-io/local-dev-cluster.git \
+		--depth=1 \
+		"$LOCAL_DEV_CLUSTER_DIR"
+}
+
+main() {
+	local op="$1"
+	shift
+
+	cd "$PROJECT_ROOT"
+	mkdir -p "${TMP_DIR}"
+
+	clone_local_dev_cluster
+	echo "Bringing cluster - $op"
+	"$LOCAL_DEV_CLUSTER_DIR/main.sh" "$op"
```

Comment:
NOTE: this allows for creation of `on_cluster_up` and `on_cluster_down` functions that can apply additional configs to the local-dev-cluster. 
E.g. https://github.com/sustainable-computing-io/kepler-operator/blob/c669624c53e5579da453e05d7254cc7c080322c6/hack/cluster.sh#L88 install OLM for `kepler-operator`

Reply:
Shouldn't we add rbac work in local-dev-cluster so that It can be used everywhere?🤔

Reply:
Sure, I would think so but that isn't the case :-). I was only pointing at an example